src/Traffic-anti-jam/Assets/Script/Vehicles/Vehicle.cs

```diff
@@ -0,0 +1,57 @@
+using UnityEngine;
+using UnityEngine.Splines;
+
+[RequireComponent(typeof(Rigidbody))]
+public class Vehicle : MonoBehaviour
+{
+    [SerializeField]
+    private SplineContainer path;
+
+    [SerializeField]
+    private Rigidbody rgbody;
+
+    [SerializeField]
+    private float speed;
+
+    private bool hasCrashed = false;
+
+    private void FixedUpdate()
+    {
+        Move();
+    }
+
+    private void OnCollisionEnter(Collision other)
+    {
+        if (other.gameObject.GetComponent<Vehicle>())
+        {
+            hasCrashed = true;
+
+            // TODO: [VD] set current game state to game over
```

Comment:
in order to keep every SOLID, this needs to fire a global event via an event bus if needed.